ambari-server/src/main/java/org/apache/ambari/server/api/services/UpgradePlanService.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ambari.server.api.services;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+import org.apache.ambari.annotations.ApiIgnore;
+import org.apache.ambari.server.api.resources.ResourceInstance;
+import org.apache.ambari.server.controller.spi.Resource;
+
+/**
+ * Endpoint for cluster upgrade plans.
+ */
+public class UpgradePlanService extends BaseService {
+
+  private String m_clusterName = null;
+
+  /**
+   * Constructor.
+   *
+   * @param clusterName the cluster name (not {@code null}).
+   */
+  UpgradePlanService(String clusterName) {
+    m_clusterName = clusterName;
+  }
+
+  @POST @ApiIgnore // until documented
+  @Produces("text/plain")
```

Comment:
1) i'm not sure, why we use `text/plain` for `application/json` content? 
2) `MediaType.TEXT_PLAIN` should be better way for usage

Reply:
Will do, thanks.  This was a copy/paste-ism.